OP-1185 | Update Inventory

ArnaudFonzam opened the pull request:
See OP-1185.
- [x] test update inventory there are requiements: 
  - [x] retrieve all products when radio button all products is clicked
  - [x]  remove the button "..."
  - [x] remove the text field next to the radio button all products
  - [x]  add new fields for movement types (charge and discharge), ward and supplier in the inventory form
  - [x] added the possibility to save inventory row without the lot
  - [x] resolve the problem about the exception during registration Inventory
  - [x]  allow multiple lots of the same product, checking there are no duplicated tuples (product, lot) upon validation
- [x] test view inventory: codeTextField and all other information related to inventory are desable and close button is visible. all others buttons are not visible the issue is to just see  the inventory row of an inventory.

Commit:
start with update issue

Commit:
update src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

Commit:
Merge branch 'OP-1184' into OP-1185

Commit:
start with update inventory

Commit:
add method to update medical inventory

Commit:
update src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

Commit:
use the good Message Dialog

Commit:
update bundle/language_en.properties

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java:
```diff
@@ -91,11 +91,11 @@ public class InventoryBrowser extends ModalJFrame implements InventoryListener {
 	JButton next;
 	JButton previous;
-	JComboBox pagesCombo = new JComboBox();
+	JComboBox<Integer> pagesCombo = new JComboBox<Integer>();
```
Why the use of objects when primitives would work????
```suggestion
	JComboBox<int> pagesCombo = new JComboBox<Integer>();
```

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -691,9 +695,12 @@ angal.inventory.specificproduct.txt
 angal.inventory.totalprice                                                                             = Total price
 angal.inventoryrow.unitprice.col                                                                       = Unit price
+angal.inventory.update.error.msg                                                                       = Inventory not updated                                       
+angal.inventory.update.succes.msg                                                                      = Inventory updated
```
```suggestion
angal.inventory.update.success.msg                                                                     = Inventory updated.
```

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -691,9 +695,12 @@ angal.inventory.specificproduct.txt
 angal.inventoryrow.theorticqty.col                                                                     = Theoritic Qty
 angal.inventory.totalprice                                                                             = Total price
 angal.inventoryrow.unitprice.col                                                                       = Unit price
+angal.inventory.update.error.msg                                                                       = Inventory not updated                                       
```
```suggestion
angal.inventory.update.error.msg                                                                       = Inventory not updated.                                       
```

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

Co-authored-by: David B Malkovsky <david@malkovsky.org>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: David B Malkovsky <david@malkovsky.org>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: David B Malkovsky <david@malkovsky.org>

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -663,15 +663,17 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventory.addlotcodebeforeenteringthecost.msg                                                    = Add lot code before entering the cost.
 angal.inventory.allproduct.txt                                                                         = All products
 angal.inventory.canceled                                                                               = Canceled
 angal.inventory.cannotcreateanotherinventorywithstatusdraft.msg                                        = Can not create another inventory with status draft.
```
```suggestion
angal.inventory.cannotcreateanotherinventorywithstatusdraft.msg                                        = Cannot create another inventory with status draft.
```

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: David B Malkovsky <david@malkovsky.org>

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -663,15 +663,17 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
 angal.inventory.inventorybrowser.title                                                                 = Inventory Browser
 angal.inventory.mustenterareference.msg                                                                = You must enter a reference.
 angal.inventory.neweditinventory.title                                                                 = Edit an inventory
+angal.inventory.noteditable.msg                                                                        = A validated inventory can not be modified.
```
```suggestion
angal.inventory.noteditable.msg                                                                        = A validated inventory cannot be modified.
```

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -691,9 +695,12 @@ angal.inventory.specificproduct.txt
 angal.inventory.validated                                                                              = Validated
 angal.inventory.validate.btn                                                                           = Validate
 angal.inventory.validate.btn.key                                                                       = V
+angal.inventory.validatednoteditable.msg                                                               = A validated inventory can not be modified.
```
```suggestion
angal.inventory.validatednoteditable.msg                                                               = A validated inventory cannot be modified.
```

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: David B Malkovsky <david@malkovsky.org>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: David B Malkovsky <david@malkovsky.org>

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -401,6 +402,54 @@ private JButton getNewButton() {
+					OHServiceExceptionUtil.showMessages(e);
+				}
+				if (checkResults == 0) {
+					MessageDialog.info(null, "angal.inventory.update.succes.msg");
```
```suggestion
					MessageDialog.info(null, "angal.inventory.update.success.msg");
```

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: David B Malkovsky <david@malkovsky.org>

ArnaudFonzam commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -168,6 +168,8 @@ private void fireInventoryUpdated() {
 	private JTextField referenceTextField;
 	private JTextField jTetFieldEditor;
 	private JButton moreData;
+	private static int MAX_COUNT = 30;
```
it can be final because it use just on line 573.

Commit:
apply changes

Commit:
update the variable type on InventoryEdit and update bundle/language_en.properties

Commit:
update src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

Commit:
add possibility to view inventory

Commit:
update bundle/language_en.properties

mwithi commented on bundle/language_en.properties:
```diff
@@ -1330,10 +1337,10 @@ angal.operationrow.not.found.msg
 angal.operationrowedit.remark                                                                          = Remarks
 angal.operationrowedit.saveerror                                                                       = Error occurs during processing
-angal.operationrowedit.savesucces                                                                      = Operation Saved!
+angal.operationrowedit.savesuccess                                                                     = Operation Saved!
```
```suggestion
angal.operationrowedit.savesuccess                                                                     = Operation Saved.
```

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

Commit:
update inventory: apply check if is AUTOMATICLOT_IN is set to yes before save

Commit:
Merge branch 'develop' into OP-1185

Commit:
remove .dist file

Commit:
update .gitignore

Commit:
revert .dist file

ArnaudFonzam committed:
Merge branch 'OP-1199-Main-Store-Inventory' into OP-1185

mwithi commented:
Note: updated https://github.com/informatici/openhospital-gui/tree/OP-1199-Main-Store-Inventory

Commit:
hidden column Lot code when AUTOMATICLOT_IN is set to yes and Prise column when LOTWITHCOST is set to no

Commit:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into OP-1185

Commit:
Merge branch 'develop' into OP-1185

ArnaudFonzam committed:
Merge branch 'informatici:develop' into OP-1185

Commit:
restore MovStockMultipleDischarging.java

Commit:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into OP-1185

Commit:
fix null pointer exception

Commit:
Merge branch 'develop' into OP-1186

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
Merge branch 'develop' into OP-1186

Commit:
add new field'

Commit:
fix all bugs

Commit:
fix bug

Commit:
update InventoryEdit.java

Commit:
add posibility to add many medical

Commit:
remove .factorypath

Commit:
add logic of view button

Commit:
add check of same lot for same medical

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
add posibility to delete multiple inventory row

Commit:
fix error index out of bounds

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -818,7 +820,12 @@ public void setValueAt(Object value, int r, int c) {
 					Lot lot = invRow.getLot();
 					if(lot == null) {
-						return ;
+						Lot lotTosotore = getLot(value.toString());
```
Fix `lotToStore`?

ArnaudFonzam replied:
yes it is to showw you the window to create lot if you cancel the action will be ended..look lines 824, 825

mwithi replied:
I mean the variable spelling: `lotToStore`, not lotTosotore

Commit:
update delete Inventory row

mwithi committed:
Merge branch 'OP-1199-Main-Store-Inventory' into OP-1185

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
update variable name

Commit:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into OP-1185

Commit:
update action to edit due date

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
add method to generate lot code

Commit:
remove method generateLotCode()

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
update InventoryEdit

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
add new column and button to store lot informations

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

Commit:
add logic on lot button

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,23 +663,42 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventory.editinventory.title                                                                    = Edit inventory
+angal.inventory.noteditable.msg                                                                        = A validated inventory cannot be modified.
 angal.inventory.notdateinfuture.msg                                                                    = A date in the future is not allowed.
 angal.inventory.noproduct.msg                                                                          = Product is not found/
```
```suggestion
angal.inventory.noproduct.msg                                                                          = Product is not found
```

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

Commit:
add status in inventory row form

Commit:
fix index out of bound exception

Commit:
add all products in inventoryrow

mwithi committed:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into ArnaudFonzam-OP-1185

mwithi committed:
Merge remote-tracking branch 'origin/OP-1199-Main-Store-Inventory' into ArnaudFonzam-OP-1185

dbmalkovsky committed:
OP-1278 resource bundle addition (#2015)

Commit:
update InventoryEdit.java

Commit:
update InventoryEdit.java

Commit:
Merge branch 'develop' into OP-1185

Commit:
fix error on bundle file

dbmalkovsky committed:
Chore: update Intellij settings to match Eclipse (#2019)

Commit:
update InventoryEdit.java

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.2.5 to 3.3.0.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.2.5...surefire-3.3.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
update InventoryEdit.java

Commit:
Merge branch 'develop' into OP-1185

dbmalkovsky committed:
Hibernate renamed org.hibernate.type to org.hibernate.orm.jdbc.bind (#2021)

Commit:
fix bug

Commit:
Merge branch 'develop' into OP-1185

Commit:
add validation on close button

Commit:
add listener on comboBox of supplier, destination, charge, discharge

Commit:
update InventoryEdit.java

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin

Bumps [org.apache.maven.plugins:maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 3.4.1 to 3.4.2.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-3.4.1...maven-jar-plugin-3.4.2)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-jar-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
fix bug related to close button

dbmalkovsky committed:
Merge pull request #2022 from informatici/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.4.2

Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin from 3.4.1 to 3.4.2

Commit:
fixed bug on InventoryEdit and InventoryBrowser

Commit:
update InventoryEdit.java

dbmalkovsky committed:
Merge pull request #2020 from informatici/dependabot/maven/org.apache.maven.plugins-maven-surefire-plugin-3.3.0

Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin from 3.2.5 to 3.3.0

Commit:
update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Commit:
fix the check of inventory properties during update mode

Commit:
change the method that fetch lot

Commit:
disable futur date on calendar

Commit:
Merge branch 'develop' into OP-1185

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
 angal.inventory.selectmedical.msg                                                                      = Select a medical.
 angal.inventory.status.txt                                                                             = Status
 angal.inventory.specificproduct.txt                                                                    = Specific product
 angal.inventoryrow.theorticqty.col                                                                     = Theoritic Qty
```
```suggestion
angal.inventoryrow.theorticqty.col                                                                     = Theoretic Qty
```

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventoryrow.pleaseselectatleastoneinventoryrow.msg                                              = Please select at least one inventory row.
+angal.inventoryrow.pleaseselectoneinventoryrow.msg                                                     = Please select one inventory row.
+angal.inventoryrow.thislotcodealreadyexists.msg                                                        = This lot code already exists. Please change it.
+angal.inventory.inventoryisalreadysave.msg                                                             = This Inventory is already Save. Please enter at least one inventory row before continue.
```
```suggestion
angal.inventory.inventoryisalreadysave.msg                                                             = This inventory is already saved. Please enter at least one inventory row before continue.
```

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventoryrow.doyoureallywanttodeletethisinventoryrow.msg                                         = Do you really want to delete this/those inventory(ies) row(s)?
+angal.inventoryrow.doyoureallywanttocleanthistable.msg                                                 = Do you really want to clean this table?
+angal.inventoryrow.doyouwanttoaddallnotyetlistedproducts.msg                                           = Do you want to add all not yet listed products?
+angal.inventoryrow.doyouwanttocontinues.msg                                                            = Do you want to continue.
```
```suggestion
angal.inventoryrow.doyouwanttocontinueediting.msg                                                            = Do you want to continue editing?
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -388,9 +565,13 @@ private JButton getNewButton() {
+							MessageDialog.info(this, "angal.inventory.savesuccess.msg");
+							fireInventoryInserted();
+							resetVariable();
+							int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinues.msg");
```
```suggestion
							int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinueediting.msg");
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -400,6 +581,212 @@ private JButton getNewButton() {
+								MessageDialog.info(null, "angal.inventory.update.success.msg");
+								resetVariable();
+								fireInventoryUpdated();
+								int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinues.msg");
```
```suggestion
								int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinueediting.msg");
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -400,6 +581,212 @@ private JButton getNewButton() {
+								MessageDialog.info(null, "angal.inventory.update.success.msg");
+								resetVariable();
+								fireInventoryUpdated();
+								int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinues.msg");
```
```suggestion
								int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinueediting.msg");
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -400,6 +581,212 @@ private JButton getNewButton() {
+					MessageDialog.info(null, "angal.inventory.update.success.msg");
+					resetVariable();
+					fireInventoryUpdated();
+					int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinues.msg");
```
```suggestion
					int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinueediting.msg");
```

mwithi replied:
I think you can extract this question into a method instead to repeat it

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
-angal.inventory.cannotsavelot.msg                                                                     = Cannot save the lot. Please verify lot information and retry.
+angal.inventory.cannotcreateanotherinventorywithstatusdraft.msg                                        = Cannot create another inventory with status draft.
+angal.inventory.cannotsavelot.msg                                                                      = Cannot save the lot. Please verify lot information and retry.
+angal.inventory.cannotsaveinventorywithoutproducts.msg                                                 = Cannot save inventory without products please insert at least one product before save.
```
```suggestion
angal.inventory.cannotsaveinventorywithoutproducts.msg                                                 = Cannot save inventory without products.
```

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventory.noteditable.msg                                                                        = A validated inventory cannot be modified.
 angal.inventory.notdateinfuture.msg                                                                    = A date in the future is not allowed.
-angal.inventory.noproduct.msg                                                                          = Product is not found/
+angal.inventory.noproduct.msg                                                                          = Product is not found
```
```suggestion
angal.inventory.noproduct.msg                                                                          = Product is not found.
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -342,32 +459,74 @@ private JButton getNewButton() {
 			if (dateInventory.isAfter(now)) {
 				MessageDialog.error(null, "angal.inventory.notdateinfuture.msg");
-				return;
+				return ;
```
And all similar occurences

```suggestion
				return;
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -342,32 +459,74 @@ private JButton getNewButton() {
+				} else {
+					inventory.setDestination(null);
+				}
 				MedicalInventory meInventory;
```
And all other occurences

```suggestion
				MedicalInventory newInventory;
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -377,7 +536,25 @@ private JButton getNewButton() {
+							} else {
 								medicalInventoryRow.setLot(null);
 							}
 							currentInventoryRow = medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
```
in which case `currentInventoryRow` could be null? if any error occurs, it will be catched, or am I wrong?

ArnaudFonzam replied:
it will be catched on catch block

mwithi replied:
exactly, so the following code should be useless, shouldn't it?

```
							if (currentInventoryRow == null) {
								checkResults++;
							}
```

ArnaudFonzam replied:
yes you right I will remove it

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -342,32 +459,74 @@ private JButton getNewButton() {
+				}
 				MedicalInventory meInventory;
 				try {
 					meInventory = medicalInventoryManager.newMedicalInventory(inventory);
```
in which case `meInventory` could be null? if any error occurs, it will be catched, or am I wrong?

ArnaudFonzam replied:
it will be catched on catch block

mwithi replied:
exactly, so the following if should be useless, shouldn't it?

```
					if (meInventory != null) {
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -408,93 +795,189 @@ private JButton getDeleteButton() {
+				if (invRows.size() == 0) {
+					selectedInventoryRow.setNewLot(true);
+					selectedInventoryRow.setLot(lot);
+					lotSaves.add(lot);
```
Refactor 

```suggestion
					lotSaved.add(lot);
```

mwithi replied:
Refactor
```suggestion
					lotsSaved.add(lot);
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -140,63 +159,94 @@ private void fireInventoryUpdated() {
+	private List<MedicalInventoryRow> inventoryRowSearchList = new ArrayList<>();
+	private List<MedicalInventoryRow> inventoryRowListAdded = new ArrayList<>();
+	private List<Lot> lotSaves = new ArrayList<>();
+	private HashMap<Integer, Lot> lotDeletes = new HashMap<>();
```
Refactor `lotDeletes` -> `lotsDeleted`

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -140,63 +159,94 @@ private void fireInventoryUpdated() {
-	private String[] pColums = { MessageBundle.getMessage("angal.common.code.txt").toUpperCase(),
+	private List<MedicalInventoryRow> inventoryRowSearchList = new ArrayList<>();
+	private List<MedicalInventoryRow> inventoryRowListAdded = new ArrayList<>();
+	private List<Lot> lotSaves = new ArrayList<>();
```
Refactor `lotSaves` -> `lotsSaved`

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -317,19 +431,22 @@ private JPanel getPanelFooter() {
 			panelFooter = new JPanel();
 			panelFooter.add(getNewButton());
```
Refactor `getNewButton` -> `getSaveButton`

Commit:
update InventoryEdit.java

Commit:
update bundle

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventoryrow.doyoureallywanttodeletethisinventoryrow.msg                                         = Do you really want to delete this/those inventory(ies) row(s)?
+angal.inventoryrow.doyoureallywanttocleanthistable.msg                                                 = Do you really want to clean this table?
+angal.inventoryrow.doyouwanttoaddallnotyetlistedproducts.msg                                           = Do you want to add all not yet listed products?
+angal.inventoryrow.doyouwanttocontinues.msg                                                            = Do you want to continue editing ?
```
```suggestion
angal.inventoryrow.doyouwanttocontinueediting.msg                                                            = Do you want to continue editing?
```

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

Commit:
remove InventoryStatus and Inventory type to medicalinventory package and update methods

Commit:
use MainStoreQuantity instead of OverAllQuantity in inventory row

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
 angal.inventory.status.txt                                                                             = Status
 angal.inventory.specificproduct.txt                                                                    = Specific product
-angal.inventoryrow.theorticqty.col                                                                     = Theoritic Qty
+angal.inventoryrow.theorticqty.col                                                                     = Theoretic Qty
```
```suggestion
angal.inventoryrow.theoreticqty.col                                                                     = Theoretic Qty
```

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventoryrow.pleaseselectatleastoneinventoryrow.msg                                              = Please select at least one inventory row.
+angal.inventoryrow.pleaseselectoneinventoryrow.msg                                                     = Please select one inventory row.
+angal.inventoryrow.thislotcodealreadyexists.msg                                                        = This lot code already exists. Please change it.
+angal.inventory.inventoryisalreadysave.msg                                                             = This inventory is already saved. Please enter at least one inventory row before continue.
```
```suggestion
angal.inventory.inventoryisalreadysaved.msg                                                             = This inventory is already saved. Please enter at least one inventory row before continue.
```

Commit:
hidden total prise when LOTWITHCOST=no

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
 angal.inventory.status.txt                                                                             = Status
 angal.inventory.specificproduct.txt                                                                    = Specific product
-angal.inventoryrow.theorticqty.col                                                                     = Theoritic Qty
+angal.inventoryrow.theoreticqty.col                                                                     = Theoretic Qty
```
Update occurences in the code also

```suggestion
angal.inventoryrow.theoreticqty.col                                                                    = Theoretic Qty
```

ArnaudFonzam replied:
I already update occurences in the code

mwithi commented on bundle/language_en.properties:
```diff
@@ -663,38 +664,72 @@ angal.hospital.thevisitinghourvaluesmustbeintherange0to24.msg
+angal.inventoryrow.pleaseselectatleastoneinventoryrow.msg                                              = Please select at least one inventory row.
+angal.inventoryrow.pleaseselectoneinventoryrow.msg                                                     = Please select one inventory row.
+angal.inventoryrow.thislotcodealreadyexists.msg                                                        = This lot code already exists. Please change it.
+angal.inventory.inventoryisalreadysaved.msg                                                             = This inventory is already saved. Please enter at least one inventory row before continue.
```
Update occurences in the code also

```suggestion
angal.inventory.inventoryisalreadysaved.msg                                                            = This inventory is already saved. Please enter at least one inventory row before continue.
```

ArnaudFonzam replied:
I already update occurences in the code

Commit:
remove useless code

ArnaudFonzam committed:
Update bundle/language_en.properties

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -140,63 +159,94 @@ private void fireInventoryUpdated() {
 			MessageBundle.getMessage("angal.wardpharmacy.lotnumber.col").toUpperCase(),
 			MessageBundle.getMessage("angal.medicalstock.duedate.col").toUpperCase(),
-			MessageBundle.getMessage("angal.inventoryrow.theorticqty.col").toUpperCase(),
+			MessageBundle.getMessage("angal.inventoryrow.theoreticqty.col").toUpperCase(),
```
look here

ArnaudFonzam commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -315,91 +429,345 @@ private JPanel getPanelContent() {
+									dispose() ; 
+								}
+							} else {
+								MessageDialog.info(null, "angal.inventory.inventoryisalreadysaved.msg");
```
look also here

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -315,91 +429,345 @@ private JPanel getPanelContent() {
-					meInventory = medicalInventoryManager.newMedicalInventory(inventory);
-					if (meInventory != null) {
-						MedicalInventoryRow currentInventoryRow;
+					newInventory = medicalInventoryManager.newMedicalInventory(inventory);
+					if (newInventory != null) {
```
Also here, if `newMedicalInventory` throws error, it will be caught, won't it?"

Commit:
remove useless code

Commit:
update inventoryEdit

Commit:
Merge branch 'OP-1185' of https://github.com/ArnaudFonzam/openhospital-gui into OP-1185

Commit:
update InventoryEdit

Commit:
use only on try block in save button

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -315,91 +429,307 @@ private JPanel getPanelContent() {
 				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+				return ;
```
Please replace all " ;" with ";"

```suggestion
				return;
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java:
```diff
@@ -315,91 +429,307 @@ private JPanel getPanelContent() {
 		saveButton = new JButton(MessageBundle.getMessage("angal.common.save.btn"));
 		saveButton.setMnemonic(MessageBundle.getMnemonic("angal.common.save.btn.key"));
 		saveButton.addActionListener(actionEvent -> {
 			String State = InventoryStatus.draft.toString();
```
Please refactor this

```suggestion
			String state = InventoryStatus.draft.toString();
```

Commit:
fix nullpointeur exception and error data has been update by someonelse

Commit:
fix error

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

ArnaudFonzam committed:
Update src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

Co-authored-by: Alessandro Domanico <alessandro.domanico@yahoo.it>

Commit:
fix error after pull

Commit:
update InventoryEdit.java

mwithi reviewed (approved):
LGTM. Thanks a lot!
